Telegraf kvm

vhe182 opened the pull request:
The goal of this PR is to add support for kvm dashboards in grafana by adding the exec plugin.  
@meteorfox @besanradwan

vhe182 committed:
playbook: Add telegraf config for compute hosts

Created a separate command for Telegraf installation on compute hosts.
Telegraf collects kvm data from compute hosts and telegraf needs an
extra plugin and file to account for this.

vhe182 committed:
group_vars: Add line for exec plugin

configure.yml needed to be changed so that the exec plugin is included
for telegraf.  The exec plugin will not be installed unless the host
falls into the compute_hosts group.

vhe182 committed:
playbook: Call to misc-perftools for compute file

Need to cause a file transfer for configuring telegraf
for compute hosts so I added a call of misc-perftools before
the call to the telegraf role.  I left the original (after the
call to telegraf) because calling "all" hosts will not trigger
the conditional I wrote for the file transfer.

I plan to either a) figure out a conditional that will trigger
even when passed all so I can move the call for misc-perftools
for "all" hosts into the call I put before the telegraf role call.
Or b) keep them separate but exclude compute_hosts from the later
misc-perftools call to minimize the redundancy.

vhe182 committed:
templates: Include telegraf.conf template

This template provides the logic so that compute_hosts can be
configured to include the exec plugin and send kvm metrics to
telegraf.

vhe182 committed:
tasks: Include directory/file creation

The kvm_virsh.py file is transfered to each compute host so a directory
needs to be created and file transfered.  I added the tasks to do this.

vhe182 committed:
playbooks: Remove redundant call to misc-perftools

The conditional I added works for when the "all" host group is
referenced.

meteorfox commented on roles/misc-perftools/tasks/main.yml:
```diff
@@ -13,6 +13,20 @@
+
+- name: Set configuration file for kvm metric collection
+  template:
+    src: /root/os-ansible-perftools/roles/misc-perftools/templates/kvm_virsh.py
```
This won't work. Don't use the template module, there's a file module. You need to put the file under the `files/` directory for this role, and just use the basename `kvm_virsh.py`. 

I'm not sure if you have to prefix the name with 'files/' or not. Here's Kevin's example.
https://github.com/openstack/openstack-ansible-ops/blob/2a1e702209cc713861bae54be2324cd2a59b3121/cluster_metrics/playbook-influx-telegraf.yml#L38-L44

vhe182 replied:
I used the copy module and it works as expected.  The example you reference uses the copy module.

meteorfox commented on roles/misc-perftools/templates/kvm_virsh.py:
```diff
@@ -0,0 +1,23 @@
+#!/usr/bin/env python
```
Please add comments indicating that this is not ours, and give credits to the author(s). If their project had a license, add the license as comment

vhe182 replied:
Added the license file and included the short version of the Apache 2.0 license to the head of kvm_virsh.py

meteorfox commented on group_vars/hosts/telegraf.yml:
```diff
@@ -48,3 +48,4 @@ telegraf_plugins:
       interfaces:
         - "bond0"
         - "bond1"
+  - name: exec
```
Where are the missing options to tell exec what to execute?

https://github.com/openstack/openstack-ansible-ops/blob/2a1e702209cc713861bae54be2324cd2a59b3121/cluster_metrics/templates/telegraf.conf.j2#L62-L65

vhe182 replied:
Thanks for catching this and clearing up my misunderstanding.  I have added the options and have tested them.

vhe182 committed:
Feedback PR-4: Added Apache 2.0 license statement

vhe182 committed:
templates: Delete file from templates directory

The file is not supposed to be in there.  It belongs in the
files directory.  Another commit places the file in the
appropriate directory.

vhe182 committed:
group_vars: Add options for telegraf config

The necessary options were not included in a previous commit.
This commit adds those options.

vhe182 committed:
tasks: Change module called (template -> copy)

I used the wrong module.  This has been corrected.

vhe182 commented:
Thanks for the feedback.  I have made all changes and the PR is ready for re-review.

meteorfox commented on roles/misc-perftools/tasks/main.yml:
```diff
@@ -13,6 +13,19 @@
+  copy:
+    src: "files/{{ item }}"
+    dest: "/opt/telegraf/{{ item }}"
+    mode: 0755
```
Add `owner` and `group` as `'telegraf'`

vhe182 replied:
okey doke.

meteorfox reviewed (changes requested):
LGTM.

Just add that minor change for `owner` and `group`.

vhe182 committed:
tasks: Add owner and group for file copy

As request in PR#4.

meteorfox reviewed (approved):
Thanks! LGTM
